tensorflow/compiler/xla/service/gpu/custom_call_test.cc

```diff
@@ -13,28 +13,51 @@ See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
 
+#if GOOGLE_CUDA
 #include "third_party/gpus/cuda/include/cuda.h"
 #include "third_party/gpus/cuda/include/cuda_runtime_api.h"
 #include "third_party/gpus/cuda/include/driver_types.h"
+#define PLATFORM "CUDA"
+#elif TENSORFLOW_USE_ROCM
+#include "rocm/include/hip/hip_runtime.h"
+#define PLATFORM "ROCM"
+#endif
 #include "tensorflow/compiler/xla/client/lib/constants.h"
 #include "tensorflow/compiler/xla/client/xla_builder.h"
 #include "tensorflow/compiler/xla/service/custom_call_target_registry.h"
 #include "tensorflow/compiler/xla/status_macros.h"
 #include "tensorflow/compiler/xla/test_helpers.h"
 #include "tensorflow/compiler/xla/tests/client_library_test_base.h"
 #include "tensorflow/core/lib/core/status_test_util.h"
+#include "tensorflow/stream_executor/gpu/gpu_types.h"
+
+#if GOOGLE_CUDA
```

Comment:
LGTM, but do you think it would be useful to have a common utility header somewhere with `ifdef`'s and `#define`'s which we can include everywhere and then not use `#if/#elif` anywhere in user code?

Reply:
> but do you think it would be useful to have a common utility header somewhere  ...

yes.  Something like the following header in the Eigen codebase
https://gitlab.com/libeigen/eigen/-/blob/master/unsupported/Eigen/CXX11/src/Tensor/TensorGpuHipCudaDefines.h

What would be the correct dir to stick such a header in? I can submit a separate PR to create that header, once I have the correct location for it.



We have some of it here
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/util/gpu_kernel_helper.h#L46-L54

Reply:
Dependencies from XLA to TF proper are a bit tricky, I think anywhere in `stream_executor` is fine.